app/src/main/java/com/google/samples/apps/sunflower/adapters/GalleryAdapter.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.adapters
+
+import android.content.Intent
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.google.samples.apps.sunflower.GalleryFragment
+import com.google.samples.apps.sunflower.data.UnsplashPhoto
+import com.google.samples.apps.sunflower.databinding.ListItemPhotoBinding
+
+/**
+ * Adapter for the [RecyclerView] in [GalleryFragment].
+ */
+
+class GalleryAdapter : ListAdapter<UnsplashPhoto, ViewHolder>(GalleryDiffCallback()) {
```

Comment:
You should be able to use `GalleryViewHolder` here instead of `ViewHolder`. Like this you get rid of the `holder as GalleryViewHolder` below.

Reply:
Done